2013-05-30  Alberto Garcia  <agarcia@igalia.com>

        [GTK] Needs to check for harfbuzz-icu
        https://bugs.webkit.org/show_bug.cgi?id=116978

        Reviewed by Xan Lopez.

        HarfBuzz 0.9.18 splits harbuzz-icu into a separate library so we
        also need to check for it in order to get the necessary flags for
        the compiler and the linker.

        We keep this conditional for now since we still want to support
        earlier versions of HarfBuzz.

        * Source/autotools/FindDependencies.m4:

Index: /trunk/Source/autotools/FindDependencies.m4
===================================================================
--- /trunk/Source/autotools/FindDependencies.m4	(revision 150962)
+++ /trunk/Source/autotools/FindDependencies.m4	(revision 150963)
@@ -376,4 +376,11 @@
         [cairo-ft fontconfig >= fontconfig_required_version freetype2 >= freetype2_required_version harfbuzz >= harfbuzz_required_version])
 fi
+# HarfBuzz 0.9.18 splits harbuzz-icu into a separate library.
+# Since we support earlier HarfBuzz versions we keep this conditional for now.
+if $PKG_CONFIG --atleast-version 0.9.18 harfbuzz; then
+    PKG_CHECK_MODULES(HARFBUZZ_ICU, harfbuzz-icu >= $harfbuzz_required_version)
+    FREETYPE_CFLAGS+=" $HARFBUZZ_ICU_CFLAGS"
+    FREETYPE_LIBS+=" $HARFBUZZ_ICU_LIBS"
+fi
 AC_SUBST([FREETYPE_CFLAGS])
 AC_SUBST([FREETYPE_LIBS])
